.tekton/tasks/save-cache.yaml

```diff
@@ -30,7 +32,10 @@ spec:
         ARTIFACTS_PATH="$(workspaces.output.path)"
         cd $ARTIFACTS_PATH
 
-        if [ ! -e "node-modules.tar.zst" ]; then
+        # we do not want to cache the prerelease node modules as we might change the node version inside container.
+        # because the node modules are stored along with branch and pre defined node version of the container.
+        # and that might broke other pipeline configurations to use the incompatible node modules
+        if [ "$(params.prerelease)" != "true" ] && [ ! -e "node-modules.tar.zst" ]; then
```

Comment:
qs: Why do we need "[ ! -e "node-modules.tar.zst" ];"?

Reply:
That condition was already there. I believe it is to prevent caching/overwriting duplicates for the same configuration.

Reply:
Ah sorry. I did not look close enough.